modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManager.java

```diff
@@ -462,7 +537,9 @@ private SCIMResponse processUserList(List<Object> tempList, JSONEncoder encoder,
             ServerSideValidator.validateRetrievedSCIMObjectInList((User) user, schema, attributes, excludeAttributes);
         }
         // Create a listed resource object out of the returned users list.
-        ListedResource listedResource = createListedResource(returnedUsers, startIndex, totalResults);
+        ListedResource listedResource = createListedResource1(returnedUsers, startIndex,
```

Comment:
createListedResource is used by listWithPOST - 1 and 2 can be handled when adjusting listWithPOST.